Refactor MCP server configurations and update Lambda function paths

esthxrs opened the pull request:
- Simplified MCP server command arguments in `.cursor/mcp.json` by removing unnecessary parameters for better clarity.
- Updated paths in `cdk/app.py` for Lambda function code assets to ensure correct referencing, enhancing deployment accuracy.
- Improved error handling and logging in `src/lambda_functions/financial_data/lambda_function.py`, optimizing the response structure for Bedrock Agent integration and adding debug statements for better traceability.
- Create portfolio_analysis

Awannaphasch2016 commented on .cursor/mcp.json:
mcp.json is OS + shell dependent. 

remove mcp.json from commit

Awannaphasch2016 commented on agent.py:
this file should be moved to isolated folder for your lambda function development.
only shared files can be in root dir.
will talk about folder structure for AWS chatbot.

Awannaphasch2016 commented on lambda_hello2.py:
remove "temp"-equivalent files. 
It could be commited by accident from me, but either way remove it.

Awannaphasch2016 commented on src/lambda_functions/financial_data/lambda_function.py:
src/lambda_functions/ folders are shared folder.

never implement test script or mock in shared files.
remove all mocking/test script from this file. 

furthermore, will discuss more on what to keep/remove/edit.

Awannaphasch2016 commented on output.json:
remove "temp"-equivalent files.

Awannaphasch2016 commented on src/lambda_functions/portfolio_analysis/certifi-2025.6.15.dist-info/INSTALLER:
we will use "poetry" to manage python pacakges.

Awannaphasch2016 commented on src/lambda_functions/portfolio_analysis/logger.py:
will impement "logger" as a shared utility function.

will discuss on this.

Awannaphasch2016 commented on trust-policy.json:
Never store files of your project in root directory.

Awannaphasch2016 commented on agent.py:
```diff
@@ -83,7 +83,7 @@ def main():
         sys.exit(1)
 
     agent_id = '52MGYIUNTN'
```
Always save these freqently changes or user-dependent variables to .env

Takedaxz committed:
Initial files

Takedaxz committed:
Add Portfolio Analysis Lambda Function and Supporting Infrastructure

- Introduced the Portfolio Analysis Lambda function, enhancing capabilities for comprehensive portfolio analysis including client overviews, performance reports, and risk metrics.
- Created supporting classes and methods for structured logging and data retrieval from Yahoo Finance.
- Established a new CDK stack for the Portfolio Analysis application, allowing for standalone deployment and integration with existing infrastructure.
- Updated requirements and added unit tests to ensure functionality and reliability of the new features.
- Removed outdated investment metrics components to streamline project structure and focus on the new portfolio analysis functionality.

Takedaxz committed:
Enhance Yahoo Finance Client with Quarterly Earnings Retrieval

- Added a new method  to the YahooFinanceClient class for fetching quarterly earnings data in PRD format.
- Implemented error handling and fallback to mock data for cases where actual data retrieval fails.
- Updated the FinancialDataService to support a new data type , integrating the quarterly earnings retrieval functionality.
- Modified the lambda function to handle  for latest earnings requests, improving compliance with PRD specifications.
- Updated .gitignore to include new directories for better environment management.

esthxrs committed:
Refactor Portfolio Analysis Lambda Function to Use User ID

- Updated the PortfolioAnalysisService to replace client_name with user_id for all portfolio analysis methods, enhancing user-specific data handling.
- Introduced a new method to retrieve user portfolio data from customer_database.csv, ensuring accurate data retrieval for analysis.
- Updated lambda_handler to extract user_id from event parameters, aligning with the new data structure.
